Added icub-gazebo-wholebody to the superbuild

gabrielenava opened the pull request:
The modification was really simple, given that the `Buildicub-gazebo-wholebody` file already exists and the `Readme` does not seem to be updated.

traversaro commented:
Thanks @gabrielenava . 
I think that it make sense to have the `icub-gazebo-wholebody` in the `DYNAMICS` profile, as it contains models used in wholebody controllers such as walking and balancing. 
Furthermore, can you add a link to it in https://github.com/robotology/robotology-superbuild#profile-cmake-options ? Thanks!

gabrielenava commented:
I modified the table describing the profile options (https://github.com/gabrielenava/robotology-superbuild/commit/ac13eb42cb430012328c3be3e5cce667dadc841d) and I modified the `CMakeLists.txt` such that both `ROBOTOLOGY_USES_GAZEBO` and `ROBOTOLOGY_ENABLE_DYNAMICS` options are required in order to have `icub-gazebo-wholebody` downloaded and installed (https://github.com/gabrielenava/robotology-superbuild/commit/283401f5d78097b3aeb8678dda1404dd744520c3).

traversaro commented on README.md:
```diff
@@ -58,7 +58,7 @@ Note that any dependencies of the included packages that is not available in the
 |:------------:|:-----------:|:---------------------:|:-------------:|:----:|
 | `ROBOTOLOGY_ENABLE_CORE` | The core robotology software packages, necessary for most users. | [`YARP`](https://github.com/robotology/yarp), [`ICUB`](https://github.com/robotology/icub-main), [`RTF`](https://github.com/robotology/robot-testing) . [`GazeboYARPPlugins`](https://github.com/robotology/GazeboYARPPlugins) and [`icub-gazebo`](https://github.com/robotology/icub-gazebo) if the `ROBOTOLOG ...
-| `ROBOTOLOGY_ENABLE_DYNAMICS` | The robotology software packages related to balancing, walking and force control. | [`iDynTree`](https://github.com/robotology/idyntree), [`WB-Toolbox`](https://github.com/robotology/WB-Toolbox), [`whole-body-controllers`](https://github.com/robotology/whole-body-controllers) | `OFF` | [Documentation on Dynamics profile.](#dynamics)  |
+| `ROBOTOLOGY_ENABLE_DYNAMICS` | The robotology software packages related to balancing, walking and force control. | [`iDynTree`](https://github.com/robotology/idyntree), [`WB-Toolbox`](https://github.com/robotology/WB-Toolbox), [`whole-body-controllers`](https://github.com/robotology/whole-body-controllers),  [`icub-gazebo-wholebody`](https://github.com/robotology-playground/icub-gazebo-wholebod ...
```
I suggest to write "[`icub-gazebo-wholebody`](https://github.com/robotology-playground/icub-gazebo-wholebody) if `ROBOTOLOGY_USES_GAZEBO` option is enabled, as in the previous line for `icub-gazebo` .

gabrielenava commented:
done with https://github.com/gabrielenava/robotology-superbuild/commit/a340f7ec5e5c63e14cb6fe0a0ce406f31f1803fe

traversaro commented on README.md:
```diff
@@ -58,7 +58,7 @@ Note that any dependencies of the included packages that is not available in the
 |:------------:|:-----------:|:---------------------:|:-------------:|:----:|
 | `ROBOTOLOGY_ENABLE_CORE` | The core robotology software packages, necessary for most users. | [`YARP`](https://github.com/robotology/yarp), [`ICUB`](https://github.com/robotology/icub-main), [`RTF`](https://github.com/robotology/robot-testing) . [`GazeboYARPPlugins`](https://github.com/robotology/GazeboYARPPlugins) and [`icub-gazebo`](https://github.com/robotology/icub-gazebo) if the `ROBOTOLOG ...
-| `ROBOTOLOGY_ENABLE_DYNAMICS` | The robotology software packages related to balancing, walking and force control. | [`iDynTree`](https://github.com/robotology/idyntree), [`WB-Toolbox`](https://github.com/robotology/WB-Toolbox), [`whole-body-controllers`](https://github.com/robotology/whole-body-controllers),  [`icub-gazebo-wholebody`](https://github.com/robotology-playground/icub-gazebo-wholebod ...
+| `ROBOTOLOGY_ENABLE_DYNAMICS` | The robotology software packages related to balancing, walking and force control. | [`iDynTree`](https://github.com/robotology/idyntree), [`WB-Toolbox`](https://github.com/robotology/WB-Toolbox), [`whole-body-controllers`](https://github.com/robotology/whole-body-controllers) and [`icub-gazebo-wholebody`](https://github.com/robotology-playground/icub-gazebo-wholeb ...
```
Sorry for the nitpick, but written like this it is not clear if `whole-body-controllers` depends or not on `ROBOTOLOGY_USES_GAZEBO`. I suggest to list all the packages included, then add a full stop, and then list the dependencies that are enabled only if `ROBOTOLOGY_USES_GAZEBO` is enabled, as in the previous line.

gabrielenava commented:
modified with https://github.com/gabrielenava/robotology-superbuild/commit/65da07db84987bbc0e5ceb6057f9886b5f20df62

traversaro commented:
Thanks!
